### ISSUE TITLE
ReadOnlyMany PVC of RBD is failed

### ISSUE THREAD

**Issue by TracyBin** [kubernetes/community#2539]
-  ReadWriteOnce is ok in the situation of relicaset is one,and pvc is mounted to only one node
- ReadOnlyMany  pvc is created successful,but deployment which mount pvc is failed
> PVC yaml
```
[admin@master1 ceph]$ cat pvc.yaml 
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: claim1
spec:
  accessModes:
    - ReadOnlyMany
  storageClassName: rbd
  resources:
    requests:
      storage: 1Gi
```
> pod example
```
[admin@master1 ceph]$ cat pod.yaml 
kind: Pod
apiVersion: v1
metadata:
  name: test-pod
spec:
  containers:
  - name: test-pod
    image: busybox:latest
    command:
    - sleep
    - "3600"
    volumeMounts:
    - name: rbdpvc
      mountPath: "/mnt"
      readonly: true
  restartPolicy: "Never"
  volumes:
  - name: rbdpvc
    persistentVolumeClaim:
      claimName: claim1
    readonly: true
```
>the  log of kubelet
```
Aug 15 17:39:10 localhost kubelet: I0815 17:39:10.302522   16197 reconciler.go:262] operationExecutor.MountVolume started for volume "pvc-a0a2a023-a06d-11e8-bbc9-005056b05298" (UniqueName: "kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1") pod "readonly-jfadmin-kubernetes-6fb8666d5-fx4ll" (UID: "0a1fa4cf-a06f-11e8-8e14-005056b02ce1")
Aug 15 17:39:10 localhost kubelet: I0815 17:39:10.302583   16197 operation_generator.go:446] MountVolume.WaitForAttach entering for volume "pvc-a0a2a023-a06d-11e8-bbc9-005056b05298" (UniqueName: "kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1") pod "readonly-jfadmin-kubernetes-6fb8666d5-fx4ll" (UID: "0a1fa4cf-a06f-11e8-8e14-005056b02ce1") DevicePath ""
Aug 15 17:39:10 localhost kubelet: I0815 17:39:10.502861   16197 operation_generator.go:455] MountVolume.WaitForAttach succeeded for volume "pvc-a0a2a023-a06d-11e8-bbc9-005056b05298" (UniqueName: "kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1") pod "readonly-jfadmin-kubernetes-6fb8666d5-fx4ll" (UID: "0a1fa4cf-a06f-11e8-8e14-005056b02ce1") DevicePath "/dev/rbd0"
Aug 15 17:39:10 localhost systemd: Started Kubernetes transient mount for /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1.
Aug 15 17:39:10 localhost systemd: Starting Kubernetes transient mount for /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1.
Aug 15 17:39:10 localhost kubelet: E0815 17:39:10.509685   16197 mount_linux.go:139] Mount failed: exit status 32
Aug 15 17:39:10 localhost kubelet: Mounting command: systemd-run
Aug 15 17:39:10 localhost kubelet: Mounting arguments: --description=Kubernetes transient mount for /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1 --scope -- mount -t ext4 -o ro /dev/rbd0 /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1
Aug 15 17:39:10 localhost kubelet: Output: Running scope as unit run-15873.scope.
Aug 15 17:39:10 localhost kubelet: mount: 文件系统类型错误、选项错误、/dev/rbd0 上有坏超级块、
Aug 15 17:39:10 localhost kubelet: 缺少代码页或助手程序，或其他错误
Aug 15 17:39:10 localhost kubelet: 有些情况下在 syslog 中可以找到一些有用信息- 请尝试
Aug 15 17:39:10 localhost kubelet: dmesg | tail  这样的命令看看。
Aug 15 17:39:10 localhost kubelet: E0815 17:39:10.509895   16197 nestedpendingoperations.go:263] Operation for "\"kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1\"" failed. No retries permitted until 2018-08-15 17:39:12.509832401 +0800 CST m=+21535.206461747 (durationBeforeRetry 2s). Error: "MountVolume.MountDevice failed for volume \"pvc-a0a2a023-a06d-11e8-bbc9-005056b05298\" (UniqueName: \"kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1\") pod \"readonly-jfadmin-kubernetes-6fb8666d5-fx4ll\" (UID: \"0a1fa4cf-a06f-11e8-8e14-005056b02ce1\") : rbd: failed to mount device /dev/rbd0 at /var/lib/kubele/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1 (fstype: ext4), error mount failed: exit status 32\nMounting command: systemd-run\nMounting arguments: --description=Kubernetes transient mount for /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1 --scope -- mount -t ext4 -o ro /dev/rbd0 /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1\nOutput: Running scope as unit run-15873.scope.\nmount: 文件系统类型错误、选项错误、/dev/rbd0 上有坏超级块、\n       缺少代码页或助手程序，或其他错误\n\n       有些情况下在 syslog 中可以找到一些有用信息- 请尝试\n       dmesg | tail  这样的命令看看
```
I think the reason is the formatting of the image in ceph. Image is automatically formatted to ext4 when in readwriteonce mode, but when PVC is readonlymany mode,Image is readonly,so format is failed

**Comment by NickrenREN** [comment]
You set `readonly: true` in your pod yaml file, kubernetes will not format the disk.

**Comment by cblecker** [comment]
Please re-post your question to [Stack Overflow](http://stackoverflow.com/questions/tagged/kubernetes).

We are trying to consolidate the channels to which questions for help/support are posted so that we can improve our efficiency in responding to your requests, and to make it easier for you to find answers to frequently asked questions and how to address common use cases.

We regularly see messages posted in multiple forums, with the full response thread only in one place or, worse, spread across multiple forums. Also, the large volume of support issues on github is making it difficult for us to use issues to identify real bugs.

The Kubernetes team scans Stack Overflow on a regular basis, and will try to ensure your questions don't go unanswered.

Before posting a new question, please search Stack Overflow for answers to similar questions, and also familiarize yourself with:

  * [user documentation](http://kubernetes.io/docs/)
  * [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/)

Again, thanks for using Kubernetes.

The Kubernetes Team

---
/triage support
/close

**Comment by TracyBin** [comment]
@NickrenREN  Thank you for your reply.But I modify readonly to False,error is exist.
```
readonly-jfadmin-kubernetes-5d489bd456-65cwb   0/1       ContainerCreating   0          4m        <none>           192.168.8.193
readonly-jfadmin-kubernetes-5d489bd456-ng7p2   0/1       ContainerCreating   0          4m        <none>           192.168.8.192
readonly-jfadmin-kubernetes-7cd848c7-2gnqt     1/1       Running             0          5m        172.20.104.7     192.168.8.191
```
The image info in node of  192.168.8.191
```
[root@191 ~]# rbd showmapped
id pool  image                                                       snap device    
0  kube1 kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1 -    /dev/rbd0 
```
PVC and pv info
```
[root@191 ~]$ kubectl get pvc
NAME            STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
readonly        Bound     pvc-cb8f0ffd-a0fc-11e8-bbc9-005056b05298   3Gi        ROX            rbd1           14m
[root@191 ~]$ kubectl get pv
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS    CLAIM                   STORAGECLASS   REASON    AGE
pvc-cb8f0ffd-a0fc-11e8-bbc9-005056b05298   3Gi        ROX            Delete           Bound     default/readonly        rbd1                     14m
```

Error log in node of 192.168.8.192,similar to 193
```
Aug 16 10:44:29 localhost kubelet: I0816 10:44:29.677281   16197 reconciler.go:217] operationExecutor.VerifyControllerAttachedVolume started for volume "pvc-cb8f0ffd-a0fc-11e8-bbc9-005056b05298" (UniqueName: "kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1") pod "readonly-jfadmin-kubernetes-5d489bd456-ng7p2" (UID: "03752d1b-a0fe-11e8-8e14-005056b02ce1")
Aug 16 10:44:29 localhost kubelet: I0816 10:44:29.680587   16197 operation_generator.go:1111] Controller attach succeeded for volume "pvc-cb8f0ffd-a0fc-11e8-bbc9-005056b05298" (UniqueName: "kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1") pod "readonly-jfadmin-kubernetes-5d489bd456-ng7p2" (UID: "03752d1b-a0fe-11e8-8e14-005056b02ce1") device path: ""
Aug 16 10:44:29 localhost kubelet: I0816 10:44:29.781288   16197 reconciler.go:262] operationExecutor.MountVolume started for volume "pvc-cb8f0ffd-a0fc-11e8-bbc9-005056b05298" (UniqueName: "kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1") pod "readonly-jfadmin-kubernetes-5d489bd456-ng7p2" (UID: "03752d1b-a0fe-11e8-8e14-005056b02ce1")
Aug 16 10:44:29 localhost kubelet: I0816 10:44:29.781343   16197 operation_generator.go:446] MountVolume.WaitForAttach entering for volume "pvc-cb8f0ffd-a0fc-11e8-bbc9-005056b05298" (UniqueName: "kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1") pod "readonly-jfadmin-kubernetes-5d489bd456-ng7p2" (UID: "03752d1b-a0fe-11e8-8e14-005056b02ce1") DevicePath ""
Aug 16 10:44:29 localhost kubelet: I0816 10:44:29.900621   16197 rbd_util.go:273] rbd image kube1/kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1 is still being used
Aug 16 10:44:29 localhost kubelet: E0816 10:44:29.900728   16197 nestedpendingoperations.go:263] Operation for "\"kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1\"" failed. No retries permitted until 2018-08-16 10:44:30.400679161 +0800 CST m=+83053.097308489 (durationBeforeRetry 500ms). Error: "MountVolume.WaitForAttach failed for volume \"pvc-cb8f0ffd-a0fc-11e8-bbc9-005056b05298\" (UniqueName: \"kubernetes.io/rbd/[192.168.8.176:6789 192.168.8.177:6789 192.168.8.178:6789]:kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1\") pod \"readonly-jfadmin-kubernetes-5d489bd456-ng7p2\" (UID: \"03752d1b-a0fe-11e8-8e14-005056b02ce1\") : rbd image kube1/kubernetes-dynamic-pvc-cb910d28-a0fc-11e8-b194-005056b02ce1 is still being used. rbd output: Watchers:\n\twatcher=192.168.8.191:0/1741855290 client.8766 cookie=4\n"
```

**Comment by TracyBin** [comment]
The document of kubernetes show that rbd pv support multiattach by nodes in the mode of ReadOnlyMany.
- ReadWriteOnce – the volume can be mounted as read-write by a single node
- ReadOnlyMany – the volume can be mounted read-only by many nodes
- ReadWriteMany – the volume can be mounted as read-write by many nodes
https://kubernetes.io/docs/concepts/storage/persistent-volumes/

**Comment by rootfs** [comment]
@TracyBin is the rbd image formatted as ext4? if not, mount just fails. The following message indicated the device was not formatted or not ext4
```
Aug 15 17:39:10 localhost kubelet: Mounting arguments: --description=Kubernetes transient mount for /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1 --scope -- mount -t ext4 -o ro /dev/rbd0 /var/lib/kubelet/plugins/kubernetes.io/rbd/rbd/kube1-image-kubernetes-dynamic-pvc-a0a516ad-a06d-11e8-b194-005056b02ce1
Aug 15 17:39:10 localhost kubelet: Output: Running scope as unit run-15873.scope.
Aug 15 17:39:10 localhost kubelet: mount: 文件系统类型错误、选项错误、/dev/rbd0 上有坏超级块、
Aug 15 17:39:10 localhost kubelet: 缺少代码页或助手程序，或其他错误
Aug 15 17:39:10 localhost kubelet: 有些情况下在 syslog 中可以找到一些有用信息- 请尝试
Aug 15 17:39:10 localhost kubelet: dmesg | tail  这样的命令看看。
```

**Comment by TracyBin** [comment]
@rootfs  because the pvc is mounted with readonly,so it can't be formatted.
I have found the reason that why the pvc can't be mounted by multi nodes with ReadOnly mode,this is a bug.The bug was fixed in release-v1.10.0,you can see it in [this](https://github.com/kubernetes/kubernetes/issues/63183)

**Comment by TracyBin** [comment]
@rootfs   I have found the reason that why pvc can't be mounted by multi nodes with ReadOnlyMany mode, this is a bug.You can see it in [this](https://github.com/kubernetes/kubernetes/issues/63183)
